src/Interpreters/InterpreterSystemQuery.cpp

```
@@ -1161,6 +1166,59 @@ void InterpreterSystemQuery::waitLoadingParts()
     }
 }
 
+
+void InterpreterSystemQuery::loadPrimaryKeys()
+{
+    if (!table_id.empty())
+    {
+        getContext()->checkAccess(AccessType::SYSTEM_LOAD_PRIMARY_KEY, table_id.database_name, table_id.table_name);
+        StoragePtr table = DatabaseCatalog::instance().getTable(table_id, getContext());
+
+        if (auto * merge_tree = dynamic_cast<MergeTreeData *>(table.get()))
+        {
+            LOG_TRACE(log, "Loading primary keys for table {}", table_id.getFullTableName());
+            try
+            {
+                merge_tree->loadPrimaryKeys();
+            }
+            catch (const Exception & ex)
+            {
+                LOG_ERROR(log, "Failed to load primary keys for table {}: {}", table_id.getFullTableName(), ex.message());
+            }
+        }
+        else
+        {
+            throw Exception(
+                ErrorCodes::BAD_ARGUMENTS, "Command LOAD PRIMARY KEY is supported only for MergeTree tables, but got: {}", table->getName());
+        }
+    }
+    else
+    {
+        getContext()->checkAccess(AccessType::SYSTEM_LOAD_PRIMARY_KEY);
+
+        /// Process databases and tables sequentially, without thread pool concurrency at the table level
+        for (auto & database : DatabaseCatalog::instance().getDatabases())
+        {
+            for (auto it = database.second->getTablesIterator(getContext()); it->isValid(); it->next())
+            {
+                if (auto * merge_tree = dynamic_cast<MergeTreeData *>(it->table().get()))
+                {
+                    try
+                    {
+                        /// Calls the improved loadPrimaryKeys in MergeTreeData
+                        merge_tree->loadPrimaryKeys();
+                    }
+                    catch (...)
+                    {
+                        LOG_ERROR(log, "Failed to load primary keys for table {}: {}", merge_tree->getStorageID().getFullTableName(), ex.message());
```

`ex` doesn't exist in this scope:

```
 1213 |                         LOG_ERROR(log, "Failed to load primary keys for table {}: {}", merge_tree->getStorageID().getFullTableName(), ex.message());
      |                                                                                                                                       ^
/home/ubuntu/ClickHouse/ClickHouse/src/Interpreters/InterpreterSystemQuery.cpp:1213:135: error: use of undeclared identifier 'ex'
```

---

Same comment about not swallowing exceptions here, we should let them bubble up.